### PR TITLE
Doc fixes

### PR DESCRIPTION
**Pull request by AlexIljin**
A few improvements.

(GitHub shows this as an old PR, but I've just pushed a bunch of stuff here, so it's supposed to be new. Not sure what went wrong, maybe the old PR was never closed when it was merged. Was it merged?)

### CODE REVIEW

File: core/syntax/syntax-docs.factor [hunk 1]
```diff
@@ -671,16 +671,16 @@ HELP: NAN:
 HELP: GENERIC:
 { $syntax "GENERIC: word ( stack -- effect )" }
 { $values { "word" "a new word to define" } }
-{ $description "Defines a new generic word in the current vocabulary. Initially, it contains no methods, and thus will throw a " { $link no-method } " error when called." } ;
+{ $description "Defines a new generic word in the current vocabulary. The word dispatches on the topmost stack element. Initially it contains no methods, and thus will throw a " { $link no-method } " error when called." } ;
```
Comment (mrjbq7) [inline core/syntax/syntax-docs.factor]: What do you mean by "dispatching on the topmost stack element"?

Comment (AlexIljin) [reply core/syntax/syntax-docs.factor]: I mean that it takes the top element on the stack, tests its class and calls the appropriate method based on the class. You know, [method dispatch](https://en.wikipedia.org/wiki/Dynamic_dispatch).

File: basis/concurrency/distributed/distributed-docs.factor [hunk 2]
```diff
@@ -14,22 +14,22 @@ ARTICLE: "concurrency.distributed.example" "Distributed Concurrency Example"
...
-  "9000 local-server <node-server> start-server drop"
+  "9000 local-server start-node"
 }
 "The code to start the thread is:"
 { $code
     "USING: concurrency.messaging threads ;"
     ": log-message ( -- ) receive . flush log-message ;"
-    "[ log-message ] \"logger\" spawn dup name>> register-remote-thread"
+    "[ log-message ] \"logger\" [ spawn ] keep register-remote-thread"
 }
```
Comment (bjourne) [inline basis/concurrency/distributed/distributed-docs.factor]: Isn't this example wrong? `register-remote-thread` has stack effect `( thread name -- )` but you pass it two thread instances.

**Comment by bjourne** [comment]
Except for the nit I had about the example in distributed-docs.factor, the PR looks good to me!

File: basis/concurrency/distributed/distributed-docs.factor (hunk 2 above)
Comment (AlexIljin) [reply basis/concurrency/distributed/distributed-docs.factor]: Umm... See the `keep` call? It keeps the "logger" string, doesn't it? Which is the thread's name... Am I right?

**Comment by mrjbq7** [comment]
Merged, thanks!!

File: basis/concurrency/distributed/distributed-docs.factor (hunk 2 above)
Comment (bjourne) [reply basis/concurrency/distributed/distributed-docs.factor]: You are right. My factor skills are getting rusty.

Comment (jonenst) [reply basis/concurrency/distributed/distributed-docs.factor]: I like
```factor
[ log-message ] "logger" [ spawn ] [ register-remote-thread ] bi
```
for symmetry but that's just style :)

Comment (AlexIljin) [reply basis/concurrency/distributed/distributed-docs.factor]: I can understand your argument, but most times I prefer shorter text, even if it saves only a couple of characters. Maybe I'll change my preference over time.

**Pull request closed without merging**